Library/Homebrew/unversioned_cask_checker.rb

```diff
@@ -115,9 +120,21 @@ def all_versions
           *colorpickers,
           *qlplugins,
           *installers,
+          *screen_savers,
         ].flat_map do |artifact|
-          source = artifact.is_a?(Cask::Artifact::Installer) ? artifact.path : artifact.source.basename
-          top_level_info_plists(Pathname.glob(dir/"**"/source/"Contents"/"Info.plist")).sort
+          sources = if artifact.is_a?(Cask::Artifact::Installer)
+            # Installers are sometimes contained within an `.app`, so try both.
+            installer_path = artifact.path
+            installer_path.ascend
+                          .flat_map { |path| (path == installer_path || path.extname == ".app") ? [path] : [] }
```

Comment:
I think this would be easier for humans to parse if it was a multi-line block rather and not using ternary operator

Reply:
Looking at this again, this could probably just be a `select`.